transformations/diverse_paraphrase/submod/submodopt.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import scipy.linalg as la
+import pdb
+
+from transformations.diverse_paraphrase.submod.submodular_funcs import (
+    distinct_ngrams,
+    ngram_overlap,
+    similarity_func,
+    seq_func,
+    ngram_overlap_unit,
+    similarity_gain,
+    seq_gain,
+)
+
+
+class SubmodularOpt:
+    def __init__(self, V=None, v=None, **kwargs):
```

Comment:
Comments here might be needed. I guess the V and v correspond to the source language and the target language?

Reply:
V is the ground set from which final candidates need to be selected.
v is the original sentence being used for selecting meaning-preserving candidates. 

As suggested, I'll add the comments.